File: Solutions/0067- Add Binary/Solution.java
class Solution {
    public String addBinary(String a, String b) {
        String shorter;
        String longer;
        StringBuilder answer = new StringBuilder();

        if (a.length() <= b.length()) {
            shorter = a;
            longer = b;
        } else {
            shorter = b;
            longer = a;
        }

        int shorterPointer = shorter.length() - 1;
        int longerPointer = longer.length() - 1;
        int carry = 0;

        while (shorterPointer >= 0) {
            if (shorter.charAt(shorterPointer) == '1' && longer.charAt(longerPointer) == '1') {
                if (carry == 0) {
                    answer.append("0");
                    carry = 1;
                } else {
                    answer.append("1");
                }
            } else if (shorter.charAt(shorterPointer) == '0' && longer.charAt(longerPointer) == '0') {
                if (carry == 0) {
                    answer.append("0");
                } else {
                    answer.append("1");
                    carry = 0;
                }
            } else {
                if (carry == 0) {
                    answer.append("1");
                } else {
                    answer.append("0");
                }
            }
            shorterPointer--;
            longerPointer--;
        }

        while (longerPointer >= 0) {
            if (longer.charAt(longerPointer) == '1') {
                if (carry == 0) {
                    answer.append("1");
                } else {
                    answer.append("0");
                }
            } else {
                if (carry == 0) {
                    answer.append("0");
                } else {
                    answer.append("1");
                    carry = 0;
                }
            }
            longerPointer--;
        }

        if (carry == 1) {
            answer.append(1);
        }

        return answer.reverse().toString();
    }
}

File: Solutions/0067- Add Binary/Solution.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        shorter, longer = (a, b) if len(a) <= len(b) else (b, a)
        shorterPointer = len(shorter) - 1
        longerPointer = len(longer) - 1
        carry = 0
        answer = []

        while shorterPointer >= 0:
            if shorter[shorterPointer] == '1' and longer[longerPointer] == '1':
                if carry == 0:
                    answer.append("0")
                    carry = 1
                else:
                    answer.append("1")
            elif shorter[shorterPointer] == '0' and longer[longerPointer] == '0':
                if carry == 0:
                    answer.append("0")
                else:
                    answer.append("1")
                    carry = 0
            else:
                if carry == 0:
                    answer.append("1")
                else:
                    answer.append("0")
            shorterPointer -= 1
            longerPointer -= 1

        while longerPointer >= 0:
            if longer[longerPointer] == '1':
                if carry == 0:
                    answer.append("1")
                else:
                    answer.append("0")
            else:
                if carry == 0:
                    answer.append("0")
                else:
                    answer.append("1")
                    carry = 0
            longerPointer -= 1

        if carry == 1:
            answer.append("1")

        return ''.join(reversed(answer))